bundles/org.openhab.binding.airgradient/src/main/java/org/openhab/binding/airgradient/internal/handler/AirGradientLocationHandler.java

```diff
@@ -0,0 +1,191 @@
+import org.openhab.binding.airgradient.internal.model.Measure;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AirGradientAPIHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jørgen Austvik - Initial contribution
+ */
+@NonNullByDefault
+public class AirGradientLocationHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(AirGradientLocationHandler.class);
+
+    public AirGradientLocationHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Channel {}: {}", channelUID, command.toFullString());
+        if (command instanceof RefreshType) {
+            Bridge bridge = getBridge();
+            if (bridge != null) {
+                AirGradientAPIHandler handler = (AirGradientAPIHandler) bridge.getHandler();
+                if (handler != null) {
+                    handler.pollingCode();
+                }
+            }
+        } else if (CHANNEL_LEDS_MODE.equals(channelUID.getId())) {
+            if (command instanceof StringType stringCommand) {
+                Bridge bridge = getBridge();
+                if (bridge != null) {
+                    AirGradientAPIHandler handler = (AirGradientAPIHandler) bridge.getHandler();
+                    if (handler != null) {
+                        handler.setLedMode(getSerialNo(), stringCommand.toFullString());
+                    }
+                }
+            } else {
+                logger.debug("Received command {} for channel {}, but it needs a string command", command.toString(),
+                        channelUID.getId());
+            }
+        } else if (CHANNEL_CALIBRATION.equals(channelUID.getId())) {
+            if (command instanceof StringType stringCommand) {
+                if ("co2".equals(stringCommand.toFullString())) {
+                    Bridge bridge = getBridge();
+                    if (bridge != null) {
+                        AirGradientAPIHandler handler = (AirGradientAPIHandler) bridge.getHandler();
+                        if (handler != null) {
+                            handler.calibrateCo2(getSerialNo());
+                        }
+                    }
+                } else {
+                    logger.debug(
```

Comment:
Not sure, but i guess this would assume an error in the users side, so let's warn him.
```suggestion
                    logger.warn(
```